docs/attestations.md

```diff
@@ -0,0 +1,76 @@
+# Attestations
+
+<!-- FIXME: this probably belongs as a docker/buildx guide -->
+
+BuildKit supports automatic creation of [in-toto attestations](https://github.com/in-toto/attestation)
+to record various pieces of metadata and information about the build.
+
+When the final output format is a container image, these attestations are
+attached to the image using the [attestation storage](./attestation-storage.md).
+
+## [SBOMs](https://en.wikipedia.org/wiki/Software_supply_chain)
+
+SBOMs are an attestation that records the software components that make up the
+final image. These consist of a list of software packages and the files that
+they own.
+
+They also usually contain metadata about each component, such as software
+licenses, authors, and unique package identifers which can be used for
+vulnerability scanning.
+
+All SBOMs generated by BuildKit are in the [SPDX](https://spdx.dev) JSON
+format. They can be generated using generator images that follow the
+[SBOM generator protocol](./sbom-protocol.md).
+
+To build an image with an attached SBOM (derived using
+[jedevc/buildkit-syft-scanner](https://github.com/jedevc/buildkit-syft-scanner)):
+
+```bash
+buildctl build \
+    --frontend=dockerfile.v0 \
+    --local context=. \
+    --local dockerfile=. \
+    --opt attest:sbom=generator=jedevc/buildkit-syft-scanner
+```
+
+### Dockerfile configuration
+
+If using the dockerfile frontend, you can also set various build arguments in
+your `Dockerfile` to more precisely control the targets that are scanned:
+
+```dockerfile
+# scan the build context using the generator
+ARG SBOM_SCAN_CONTEXT=true
+
+FROM alpine:latest as build
+# scan the the build stage using the generator
+ARG SBOM_SCAN_TARGET=true
+RUN ... # build some software
+
+FROM scratch
+# the final stage is always scanned by default
+COPY --from=build /path/to/software /path/to/software
+```
+
+If the generator image passed as an option in `attest:sbom` supports it, an
+additional SBOM for the build context and each target will be generated in
+addition to an SBOM for the final stage.
+
+You can also directly override these `ARG`s on the command line, by passing
+them as build-arguments.
+
+```bash
+buildctl build \
+    --frontend=dockerfile.v0 \
+    --local context=. \
+    --local dockerfile=. \
+    --opt build-arg:SBOM_SCAN_STAGE=true \
+    --opt build-arg:SBOM_SCAN_CONTEXT=true \
+    --opt attest:sbom=generator=jedevc/buildkit-syft-scanner
+```
+
```

Comment:
We could show a snippet of how the SPDX content actually looks like.

How to add custom packages to SBOM using the nested SBOM technique.

How to read the layer info from the SPDX record.

Reply:
1. Added an SPDX snippet :heavy_check_mark: 
2. This is scanner-specific behaviour - we should document this in the scanner itself, since a scanner may not include this information. Also, this behaviour isn't fully implemented in syft, and is partially blocked on https://github.com/anchore/syft/pull/1029. :heavy_multiplication_x:
3. Added info on the layer data :heavy_check_mark:

Reply:
CC @wagoodman @kzantow :tada:

Reply:
Does the default scanner support the nested files now that the PR looks to be merged?

I think this should be documented in the regular sbom docs like the Dockerfile ARGs. This is something that we expect the users to know and modify their Dockerfiles to get better sboms. We can mention that this works in default scanner and if user uses custom scanner then it is up to that implementation to decide if it is supported.

Reply:
Ahh, so yes syft has merged it, but it's not made into a release.

I think we should aim to follow syft releases for buildkit-syft-scanner, instead of building off of master?

Reply:
We already vendor a syft release right? https://github.com/docker/buildkit-syft-scanner/blob/6a7324fe9deac77f3a6f22770b7a8a09b4ce1723/go.mod#L8